meetings/2021-08/sept-01.md

```diff
@@ -0,0 +1,585 @@
+
+JHD: And then I have another open question here. It’s just about: can we, or should we, at all restrict hosts around what `getIntrinsic` function they can provide. But these are all things that I’d like to know—these are things to explore within Stage 1. And I like to spend minimal time, so we can finish the timebox sonner discussing those today. So that’s my spiel.
+
+SYG: I’ll start with the implementation concerns. So there was a bit you said about, you know, how big the user-land library is. For V8, at least the cost is, I mean, it’s not going to be free. There will be non-trivial memory cost because…while it is true that some built-ins are held onto what’s called a native context in the V8 codebase, which is basically like the thing that holds onto globals, including non-user exposed ones…While it’s true that some built-ins are there, it is not the case that close to most, or all, of them are on there. And to add a bunch of references as would be required is very undesirable. And this was the problem that last time `getIntrinsic` came up.
+
+SYG: And last time there was a solution that was proposed that required architecture: which is, you know, [that] probably V8 should move to some kind of lazy-loading thing. Anyway, instead of trying to have everything on the global to begin with. And that’s still probably the best chance going forward to recoup the memory costs—to not punish, you know, every every context. The thing I would like to caution is that this approach means that this might not get implemented in a timely fashion, because the re-architecture is significant. But, that is to say, the implementation concerns still exist. Independently [the] lazy-loading thing is probably good anyway for the codebase to do, but I can’t promise any kind of timelines there. I know this is just Stage 1, but, right, I just wanted to set expectations.
+
+JHD: And I’ll just say that if the, if implementations as a group are confident they can eventually ship it and they plan to and that they have no, you know, and as long as some implementations or implementations can ship it obviously because of the requirements for advancing stage for I’m comfortable with the personally, because once it’s part of the specification, I can build a polyfill and then that thing can just fall out of usage naturally over time. So I’m  thinking about, you know, the next ten years. Not the next you know, ten months. So it’s totally fine to me if there’s a delay, but thank you for sharing that concern.
+
+JWK: So, I have a question for the engine. Like, XS’s engine needs to be small. And what if, and I think that `getIntrinsic` is good. But it seems like we need to add too many strings into the engine, because those intrinsics are created by the strings like that. I guess that might add too much size to the embedded JS engines like XS. Maybe we can only add intrinsics that can only be reached by syntax. To get the list smaller. 
```

Comment:
```suggestion
JWK: So, I have a question for an engine like XS. They need to be small. I think `getIntrinsic` is good. But it seems like we need to add too many strings into the engine because those intrinsics are created by the strings like `%ArrayPrototype%.slice`. I guess that might add too much size to the XS. Maybe we can only add intrinsics that can only be reached by syntax to get the list smaller. 
```